File: Longest Consecutive Sequence.py
#https://www.naukri.com/code360/problems/longest-consecutive-sequence_759408

from math import *
from collections import *
from sys import *
from os import *

def lengthOfLongestConsecutiveSequence(arr, n):
    # Write your code here.
    arr=sorted(arr)
    result=1
    count=1
    for i in range(1,n):
        if arr[i]-arr[i-1]==1:
            count+=1
        elif arr[i]==arr[i-1]:
            pass
        else:
            count=1
        result=max(result,count)
    return result

File: Longest Consecutive Sequence.cpp
//https://leetcode.com/problems/longest-consecutive-sequence/



class Solution {
public:
    int result=0;
    unordered_map<int,int> map; 
    int find(int x){
        if(!map.count(x)) return x+1;
        else if(map[x]==-1) map[x]=find(x-1);
        return map[x]; 
    }
    void uni(int x, int y){
        if(map.count(y) && map[y]!=-1) {
            map[x]=map[y];
            result=max(result,x-map[x]);
            return;
        }
        // if(!map.count(y)) map[x]=x;
        else{
            int f=find(y);
            map[x]=f;
            result=max(result,x-map[x]);
        }

    }
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0) return 0;
        for(int i=0;i<nums.size();i++) map[nums[i]]=-1;

        for(int i:nums)
            uni(i,i-1);
        // for(auto p:map){
        //     cout<<p.first<<" "<<p.second<<endl;
        // }
        return result+1;
    }
};
